src/south/SQLDbToFile/SQLDbToFile.class.js

```diff
@@ -101,35 +121,44 @@ class SQLDbToFile extends ProtocolHandler {
     entryList.forEach((entry) => {
       if (entry[this.timeColumn] instanceof Date && entry[this.timeColumn] > newLastCompletedAt) {
         newLastCompletedAt = entry[this.timeColumn]
-      } else if (entry[this.timeColumn] && new Date(entry[this.timeColumn]).toString() !== 'Invalid Date') {
-        const entryDate = new Date(moment.tz(entry[this.timeColumn], this.timezone).tz('UTC').toISOString())
-        if (entryDate > newLastCompletedAt) {
-          newLastCompletedAt = entryDate
+      } else if (entry[this.timeColumn]) {
+        const entryDate = new Date(entry[this.timeColumn])
+        if (entryDate.toString() !== 'Invalid Date') {
+          // When of type string, We need to take back the js added timezone since it is not in the original string coming from the database
+          // When of type number, no need to take back the timezone offset because it represents the number of seconds from 01/01/1970
+          // eslint-disable-next-line max-len
+          const entryDateWithoutTimezoneOffset = typeof entry[this.timeColumn] === 'string' ? new Date(entryDate.getTime() - entryDate.getTimezoneOffset() * 60000) : entryDate
+          if (entryDateWithoutTimezoneOffset > new Date(newLastCompletedAt)) {
+            newLastCompletedAt = entryDateWithoutTimezoneOffset
+          }
         }
       }
     })
 
     if (newLastCompletedAt !== actualLastCompletedAt) {
-      this.logger.debug(`Updating lastCompletedAt to ${newLastCompletedAt.toISOString()}`)
+      this.logger.debug(`Updating lastCompletedAt to ${newLastCompletedAt}`)
     } else {
       this.logger.debug('lastCompletedAt not used')
     }
     return newLastCompletedAt
   }
 
   /**
-   * Get entries from the database since the last query completion, write them into a CSV file and send to the Engine.
-   * @param {*} scanMode - The scan mode
+   * Get entries from the database since the last query completion,
+   * write them into a CSV file and send to the Engine.
+   * @param {String} scanMode - The scan mode
+   * @param {Date} startTime - The start time
+   * @param {Date} endTime - The end time
    * @return {void}
    */
-  async onScanImplementation(scanMode) {
+  async historyQuery(scanMode, startTime, endTime) {
     if (!this.timezone) {
       this.logger.error('Invalid timezone')
       return
     }
 
-    if (this.ongoingReads[scanMode]) {
-      this.logger.silly(`onScan ignored: ongoingReads[${scanMode}]: ${this.ongoingReads[scanMode]}`)
+    if (!((this.nrOfReplacements === 0) || (this.nrOfReplacements === 3))) {
+      this.logger.error('Invalid query format. Please use all or nothing from @StartTime, @EndTime, @MaxReturnValues')
       return
     }
```

Comment:
After we discussed about the migration steps, I think we should allow to use only some of these variables. Particularly StartTime only so it will be easier to automatically migrate LastCompletedDate to StartTime